drop-transfer/src/ws/server/mod.rs

```diff
@@ -780,3 +712,111 @@ fn move_tmp_to_dst(
 
     Ok(dst_location)
 }
+
+async fn init_client_handler(
+    state: &State,
+    xfer: &crate::Transfer,
+    req_send: mpsc::UnboundedSender<ServerReq>,
+    logger: &Logger,
+) -> anyhow::Result<()> {
+    state
+        .transfer_manager
+        .lock()
+        .await
+        .insert_transfer(xfer.clone(), TransferConnection::Server(req_send.clone()))
+        .context("Failed to insert a new transfer")?;
+
+    let existing_info = match state.storage.incoming_transfer_info(xfer.id()).await {
+        Ok(info) => info,
+        Err(err) => {
+            error!(
+                logger,
+                "DB failed upon checking if transfer is a resume: {err:?}",
+            );
+            // Fall back to treating it as a new one
+            None
+        }
+    };
+
+    let current_info = xfer.storage_info();
+    if let Some(existing_info) = existing_info {
+        ensure_resume_matches_existing_transfer(current_info, existing_info)
+            .with_context(|| format!("Failed to validate transfer {} resume", xfer.id()))?;
+
+        info!(
+            logger,
+            "Transfer {} resume. Resuming started files",
+            xfer.id()
+        );
+
+        if let Err(err) = resume_transfer_files(state, xfer, &req_send, logger).await {
+            warn!(logger, "Failed to resume started files: {err:?}");
+        }
+    } else {
+        if let Err(err) = state.storage.insert_transfer(&current_info).await {
+            error!(logger, "Failed to insert transfer into the DB: {err:?}");
+        }
+
+        state
+            .event_tx
+            .send(Event::RequestReceived(xfer.clone()))
+            .await
+            .expect("Failed to notify receiving peer!");
+    }
+
+    Ok(())
+}
```

Comment:
It's clearer now to see that when we receive a transfer we match it with what exists in the database, and if it's _exactly_ the same we automatically initiate the resume while omitting `TransferRequested` event.

Two points that stand out:
- we use the unbounded channel, thus if our transfer contains 1000 files for example, we will initiate 1000 parallel downloads with no rate-limiting
- Why do we use a channel to communicate transfer tasks but we also have `transfer_manager`, it looks like two detached communication channels. Maybe it's worth wrapping the channel into TransferManager?

Reply:
Yes, we do not have a rate limit, but it's the same for the app `download` calls. Actually, the number of files to restart here is exactly what the app allowed on the previous run.
Regarding the `transfer_manager`, the same chancel is stored there. I've extracted it only to not clutter these methods with manager calls which is additionally mutex protected

Reply:
The app at least controls how many `download()` calls it issues. 

Consider where the app tries one by one to download 100 files and all fail. After restarting all 100 will be in flight.
It would be nice to rate-limit to some sane number, like providing a configuration limit with a default of 16 for example.

The problem with that though is that it might race with other API calls given that we queue.